src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -0,0 +1,302 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useImperativeHandle, useState } from 'react';
+import { useContainerQuery } from '@cloudscape-design/component-toolkit';
+import { useControllable } from '../../internal/hooks/use-controllable';
+import { fireNonCancelableEvent } from '../../internal/events';
+import { useFocusControl } from '../utils/use-focus-control';
+import { useMobile } from '../../internal/hooks/use-mobile';
+import { useSplitPanelFocusControl } from '../utils/use-split-panel-focus-control';
+import { useDrawers } from '../utils/use-drawers';
+import { useUniqueId } from '../../internal/hooks/use-unique-id';
+import { AppLayoutProps, AppLayoutPropsWithDefaults } from '../interfaces';
+import { computeHorizontalLayout, computeVerticalLayout } from './compute-layout';
+import { SplitPanelProviderProps } from '../split-panel';
+import { SplitPanelSideToggleProps } from '../../internal/context/split-panel-context';
+import { getSplitPanelDefaultSize } from '../../split-panel/utils/size-utils';
+import globalVars from '../../internal/styles/global-vars';
+import { SkeletonLayout } from './skeleton';
+import {
+  AppLayoutDrawer,
+  AppLayoutNavigation,
+  AppLayoutNotifications,
+  AppLayoutSplitPanelBottom,
+  AppLayoutSplitPanelSide,
+  AppLayoutToolbar,
+} from './internal';
+import { AppLayoutInternals } from './interfaces';
+
+const AppLayoutVisualRefreshToolbar = React.forwardRef(
+  (
+    {
+      ariaLabels,
+      contentHeader,
+      content,
+      navigationOpen,
+      navigationWidth,
+      navigation,
+      navigationHide,
+      onNavigationChange,
+      tools,
+      toolsOpen: controlledToolsOpen,
+      onToolsChange,
+      toolsHide,
+      toolsWidth,
+      contentType,
+      headerVariant,
+      breadcrumbs,
+      notifications,
+      stickyNotifications,
+      splitPanelPreferences: controlledSplitPanelPreferences,
+      splitPanelOpen: controlledSplitPanelOpen,
+      splitPanel,
+      splitPanelSize: controlledSplitPanelSize,
+      onSplitPanelToggle,
+      onSplitPanelResize,
+      onSplitPanelPreferencesChange,
+      disableContentPaddings,
+      minContentWidth,
+      maxContentWidth,
+      placement,
+      ...rest
+    }: AppLayoutPropsWithDefaults,
+    forwardRef: React.Ref<AppLayoutProps.Ref>
+  ) => {
+    const isMobile = useMobile();
+    const embeddedViewMode = (rest as any).__embeddedViewMode;
+    const splitPanelControlId = useUniqueId('split-panel');
+    const [toolbarState, setToolbarState] = useState<'show' | 'hide'>('show');
+
+    const onNavigationToggle = (open: boolean) => {
+      fireNonCancelableEvent(onNavigationChange, { open });
```

Comment:
I think the focus handling is missing here. When closing the navigation, the focus should move to the open button and vice versa. Can be reproduced [on this page](https://d21d5uik3ws71m.cloudfront.net/components/97753245901d23869c4530abadd065aa4b936694/dev-pages/index.html#/light/app-layout/runtime-drawers/?appLayoutWidget=true) when using the keyboard navigation to toggle the navigation.

The same focus issue applies for the tools toggle button behaviour.

Reply:
This code is still work in progress and focus handling is not in the scope of this PR. All related tests are also skipped for the same reason